spring-pulsar-test/src/main/java/org/springframework/pulsar/test/support/ConditionsStep.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.pulsar.test.support;
+
+import java.time.Duration;
+import java.util.List;
+
+import org.apache.pulsar.client.api.Message;
+
+/**
+ * Assertions related step in the fluent API for building a Pulsar test consumer.
+ *
+ * @param <T> the type of the message payload
+ * @author Jonas Geiregat
+ */
+public interface ConditionsStep<T> {
```

Comment:
I like the introduction of the "Step" as it gives direction on what operations leave to what, rather than stuffing all into the builder and relying on docs to point the user to how to use it exactly. I think using "Spec" instead of "Step" lends itself better to the builder paradigm though.

Reply:
I chose `Step` as the idea is derived from the `StepBuilder` pattern.